fe/fe-core/src/main/java/com/starrocks/qe/GlobalVariable.java

```diff
@@ -87,6 +93,89 @@ public final class GlobalVariable {
     @VariableMgr.VarAttr(name = "performance_schema", flag = VariableMgr.READ_ONLY)
     private static boolean performanceSchema = false;
 
+    /**
+     * Query will be pending when BE is overloaded, if `queryQueueEnable` is true.
+     * <p>
+     * If the number of running queries of any BE `exceeds queryQueueConcurrencyHardLimit`,
+     * or memory usage rate of any BE exceeds `queryQueueMemUsedPctHardLimit`,
+     * the current query will be pending or failed:
+     * - if the number of pending queries in this FE exceeds `queryQueueMaxQueuedQueries`,
+     * the query will be failed.
+     * - otherwise, the query will be pending until all the BEs aren't overloaded anymore
+     * or timeout `queryQueuePendingTimeoutSecond`.
+     * <p>
+     * Every BE reports at interval the resources containing the number of running queries and memory usage rate
```

Comment:
how memory usage rate is computed out？ is it an instant quantity or an aging sliding statistics quantity, I think the latter choice is better, the query queue should not be too sensitive to memory usage varying. It should find out some periodic overloading pattern to avoid.

Reply:
Maybe we should use instant value, since the latency of collecting resource statistics is expected.